client/src/components/AuthComponent.jsx

```diff
@@ -20,33 +21,39 @@ import {
 import { signInWithPopup } from 'firebase/auth';
 import React, { useContext, useState } from 'react';
 import { FcGoogle } from 'react-icons/fc';
-import { useNavigate } from 'react-router-dom';
+import { Link as RouterLink, useNavigate } from 'react-router-dom';
 import secureLocalStorage from 'react-secure-storage';
 
 import AuthContext from '../contexts/AuthContext';
 import { auth, provider } from '../utils/firebaseConfig';
 import ErrorMessage from './ErrorMessage';
 
-const AuthComponent = () => {
+const AuthComponent = ({ isSignInPage }) => {
   let navigate = useNavigate();
+  const validEmail =
```

Comment:
There is a lot of authentication logic housed in the same logic that handles rendering of the page itself.

Could we move a bunch of the authentication into it's own component, and then import the functions we need into the sign in component? Or could we move the front-end elements into `SignUpPage.jsx` or `SignInPage.jsx`?